4 - AnimalShelter/Code/Enums/SupportedAnimals.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AnimalShelter.Code.Enums
+{
+    public enum SupportedAnimals
+    {
+        cat,
+        dog,
+        bird,
+        snake,
+    }
```

Comment:
Though they share the same value types in spelling, your supported animals is NOT directly compatible with with your animals enum. Due to this is why you are having to use strings and find unique ways to go from one enum to another.

To get around this, a section of code could simply have a list of enum values form the animals and the variable is called "supported". Another approach is to mark the enum value's with attribitues to then use reflection to get a list at runtime with your supported animals.